plugin/mutate/source/dextool/plugin/mutate/backend/database/standalone.d

```diff
@@ -280,12 +280,63 @@ struct Database {
         return app.data.sort!((a, b) => a.timeStamp < b.timeStamp).array;
     }
 
+    /// Returns: the stored scores in ascending order by their `time`.
+    FileScore[] getMutationFileScoreHistory() @trusted {
+        import std.algorithm : sort;
+
+        auto app = appender!(FileScore[])();
+        foreach (r; db.run(select!MutationFileScoreHistoryTable)) {
+            app.put(FileScore(r.timeStamp, typeof(FileScore.score)(r.score), Path(r.filePath)));
+        }
+
+        return app.data.sort!((a, b) => a.timeStamp < b.timeStamp).array;
+    }
+
     /// Add a mutation score to the history table.
     void putMutationScore(const MutationScore score) @trusted {
         db.run(insert!MutationScoreHistoryTable, MutationScoreHistoryTable(0,
                 score.timeStamp, score.score.get));
     }
 
+    // Add a mutation score for the individual files
+    void putFileScore(const FileScore score) @trusted {
+        db.run(insert!MutationFileScoreHistoryTable, MutationFileScoreHistoryTable(0,
+                score.timeStamp, score.score.get, score.file.toString));
+    }
+
+    void removeFileScores() @trusted {
+        auto stmt = db.prepare(
+                "DELETE FROM " ~ mutationFileScoreHistoryTable ~ " 
+                    WHERE file_path NOT IN (
+                    SELECT DISTINCT path
+                    FROM " ~ filesTable ~ ")");
+        stmt.get.execute;
+    }
+
+    void trimFileScore(const long keep, Path file) @trusted {
+        auto stmt = db.prepare(format!"SELECT count(*) FROM %s WHERE file_path=:file"(mutationFileScoreHistoryTable));
+        stmt.get.bind(":file", file.toString);
+        const sz = stmt.get.execute.oneValue!long;
+
+        if (sz < keep)
+            return;
+
+        auto ids = appender!(long[])();
+        stmt = db.prepare(format!"SELECT id FROM %s WHERE file_path=:file ORDER BY time_stamp ASC LIMIT :limit"(
```

Comment:
I think for sqlite to be able to order datetime columns correctly you need to use `datetime(time_stamp)` for it to understand it is "time" and not something else.